src/Integration.Vsix/ErrorList/IssuesSnapshot.cs

```diff
@@ -240,5 +256,46 @@ public override int IndexOf(int currentIndex, ITableEntriesSnapshot newSnapshot)
             }
             return base.IndexOf(currentIndex, newSnapshot);
         }
+
+        #endregion
+
+        #region IIssuesSnapshot implementation
+
+        public Guid AnalysisRunId { get; }
+
+        public IEnumerable<IssueMarker> IssueMarkers => readonlyIssueMarkers;
+
+        public IEnumerable<string> FilesInSnapshot { get; }
+
+        public IEnumerable<IAnalysisIssueLocationVisualization> GetLocationsVizsForFile(string filePath)
+        {
+            if (!FilesInSnapshot.Any(x => IsMatchingPath(filePath, x)))
+            {
+                return Array.Empty<IAnalysisIssueLocationVisualization>();
+            }
+
+            var locVizs = EnumerateAllLocationVisualizations(IssueMarkers)
+                .Where(locViz => IsMatchingPath(filePath, locViz.Location.FilePath));
+
+            return new List<IAnalysisIssueLocationVisualization>(locVizs);
+        }
+
+        private static IEnumerable<string> CalculateFilesInSnapshot(IEnumerable<IssueMarker> markers)
+        {
+            var allFilePaths = EnumerateAllLocationVisualizations(markers)
+                .Select(locViz => locViz.Location.FilePath);
+
+            return new HashSet<string>(allFilePaths, StringComparer.OrdinalIgnoreCase);
+        }
+
+        private static IEnumerable<IAnalysisIssueLocationVisualization> EnumerateAllLocationVisualizations(IEnumerable<IssueMarker> markers) =>
```

Comment:
Maybe 'GetAllLocationVisualizations' ? `Enumerate` implies to me that you receive a callback to do something with the enumeration. `Get` is more accurate IMO.